Grid.hpp

```diff
@@ -208,39 +195,43 @@ class LIB_EXPORT Grid {
     Grid(MPI_Comm comm, const std::string& filename, const std::string& dim0_id = "x",
-        const std::string& dim1_id = "y", const std::string& mask_id = "mask", int blk0 = 1,
-        int blk1 = 1, bool ignore_mask = false);
+        const std::string& dim1_id = "y", const std::string& mask_id = "mask",
+        bool ignore_mask = false);
+
+    /*!
+     * @brief Read dims from netcdf grid file.
+     *
+     * @param filename filename of the input netcdf grid file.
+     */
+    void ReadGridDims(const std::string& filename);
+
+    /*!
+     * @brief Read data from netcdf grid file.
+     *
+     * @param filename filename of the input netcdf grid file.
+     * @param mask_name name of the land mask in the grid file.
+     */
+    void ReadGridMask(const std::string& filename, const std::string& mask_name);
 
 private:
     MPI_Comm _comm; // MPI communicator
     int _rank = -1; // Process rank
     int _num_procs = -1; // Total number of processes in communicator
     int _num_procs_0 = -1; // Total number of processes in 1st dimension
     int _num_procs_1 = -1; // Total number of processes in 2nd dimension
-    int _blk_factor_0 = 1; // Blocking factor in 1st dimension
-    int _blk_factor_1 = 1; // Blocking factor in 2nd dimension
     int _global_ext_0 = 0; // Global extent in 1st dimension
     int _global_ext_1 = 0; // Global extent in 2nd dimension
-    int _global_ext_blk_0 = 0; // Global extent in 1st dimension (blocking)
-    int _global_ext_blk_1 = 0; // Global extent in 2nd dimension (blocking)
     int _local_ext_0 = 0; // Local extent in 1st dimension
     int _local_ext_1 = 0; // Local extent in 2nd dimension
-    int _local_ext_blk_0 = 0; // Local extent in 1st dimension (blocking)
-    int _local_ext_blk_1 = 0; // Local extent in 2nd dimension (blocking)
     int _global_0 = -1; // Upper left corner global coordinate in 1st dimension
     int _global_1 = -1; // Upper left corner global coordinate in 2nd dimension
-    int _global_blk_0 = -1; /* Upper left corner global coordinate in 1st
-                               dimension (blocking) */
-    int _global_blk_1 = -1; /* Upper left corner global coordinate in 2nd
-                               dimension (blocking) */
     int _num_objects = 0; // Number of grid points ignoring land mask
-    int _num_blks = 0; // Number of grid blocks ignoring land mask
     int _num_nonzero_objects = 0; // Number of non-land grid points
-    int _num_nonzero_blks = 0; // Number of non-land grid blocks
-    std::vector<int> _land_mask = {}; // Land mask values of grid bGpoints
-    std::vector<int> _land_mask_blk = {}; // Land mask values of grid blocks
-    std::vector<int> _sparse_to_dense = {}; // Map from sparse to dense index of blocks
-    std::vector<int> _object_id = {}; // Unique non-land grid block IDs
+    const std::string& _dim0_name; // dim0 name
+    const std::string& _dim1_name; // dim1 name
```

Comment:
The comments here aren't particularly helpful.

Reply:
agreed but they are wiped in #58. So I won't bother changing them here.